src/main/java/DungeonController.java

```diff
@@ -0,0 +1,242 @@
+                }
+            }
+        }
+    }
+
+    private void addPassage(Dungeon dungeon) {
+        Random rand = new Random();
+        int doorLocation;
+
+        //Top vertical door
+        doorLocation = rand.nextInt((dungeon.getHeight() / 2 - 1) - 1) + 1;
+        gridBounds[doorLocation][dungeon.getWidth() / 2] = 0;
+        dungeonMatrix[doorLocation][dungeon.getWidth() / 2] = " ";
+
+        //Bottom vertical door
+        doorLocation = rand.nextInt((dungeon.getHeight() - 1) - (dungeon.getHeight() / 2 + 1)) + (dungeon.getHeight() / 2 + 1);
+        gridBounds[doorLocation][dungeon.getWidth() / 2] = 0;
+        dungeonMatrix[doorLocation][dungeon.getWidth() / 2] = " ";
+
+        //Left horizontal door
+        doorLocation = rand.nextInt((dungeon.getWidth() / 2 - 1) - 1) + 1;
+        gridBounds[dungeon.getHeight() / 2][doorLocation] = 0;
+        dungeonMatrix[dungeon.getHeight() / 2][doorLocation] = " ";
+
+        //Right horizontal door
+        doorLocation = rand.nextInt((dungeon.getWidth() - 1) - (dungeon.getWidth() / 2 + 1)) + (dungeon.getWidth() / 2 + 1);
+        gridBounds[dungeon.getHeight() / 2][doorLocation] = 0;
+        dungeonMatrix[dungeon.getHeight() / 2][doorLocation] = " ";
+    }
+
+    private void addExit(Dungeon dungeon) {
+        int exitLocation;
+        int wall;
+        Random rand = new Random();
+        wall = rand.nextInt(4); //Random integer between 1 and 4
+        //Uses height only to find middle dungeon values, will not work if not a square
+        exitLocation = rand.nextInt((dungeon.getHeight() - 1) - 1) + 1;
+        while (exitLocation == dungeon.getHeight() / 2) {
+            exitLocation = rand.nextInt((dungeon.getHeight() - 1) - 1) + 1;
+            System.out.println("Moved Exit");
+        }
+
+        switch (wall) {
+            case 0: //Top wall
+                dungeonMatrix[0][exitLocation] = "E";
+                gridBounds[0][exitLocation] = 5;
+                break;
+            case 1: //Bottom wall
+                dungeonMatrix[dungeon.getHeight() - 1][exitLocation] = "E";
+                gridBounds[dungeon.getHeight() - 1][exitLocation] = 5;
+                break;
+            case 2: // Left wall
+                dungeonMatrix[exitLocation][0] = "E";
+                gridBounds[exitLocation][0] = 5;
+                break;
+            case 3: //Right wall
+                dungeonMatrix[exitLocation][dungeon.getWidth() - 1] = "E";
+                gridBounds[exitLocation][dungeon.getWidth() - 1] = 5;
+                break;
+        }
+    }
+
+    private void addRoomBounds(Dungeon dungeon, int rooms) {
+        Random rand = new Random();
+        int startHeight;
+        int startWidth;
+
+        for (int r = 1; r <= rooms; r++) { //1 - top left, 2 - top right, 3 - bottom left, 4 - bottom right
+
+            switch (r) {
+                case 1:
+                    startHeight = rand.nextInt((dungeon.getHeight() / 2 - 6) - 2) + 2;
```

Comment:
can you declare ints 2 and 6 to make more readable and only need changing in one place - assuming they are max and min